### DISCUSSION TITLE
Interplay between the Phase Field module and the Tensor Mechanics module

### DISCUSSION THREAD

**Discussion by RobMacka** [idaholab/moose#25615] [Q&A Modules: General]
Hello all, 

I'm writing this in the general section as it covers both the Phase Field and the Tensor Mechanics modules. I'm looking to implement strain fields using inclusion defects with the Tensor Mechanics module, while also evolving the strain with an equation under the Phase Field model which includes strain components as a full variable. However, since the Tensor Mechanics module outputs the strain tensor components as AuxVariables, I can't evolve these under the Allen-Cahn kernel as it is incompatible with AuxVariables. Is there a way around this i.e. to cast the strains from the Tensor Mechanics module as full variables, or is this not possible within the MOOSE framework? Thank you for your time

**Comment by GiudGiud** [comment]
Hello

> Tensor Mechanics module outputs the strain tensor components as AuxVariables

this output is just for visualization purposes.
For coupling purposes, use the underlying material property
There should be some work on PF - TM coupling in the combined module you can look at. 

Guillaume

**Reply by RobMacka** [reply]
Hello Guillaume,

Thank you for your answer, sorry I don't think I explained the problem properly. The Phase Field problem i wish to evolve depends on different variables unrelated to that causing the strain in the Tensor Mechanics problem (basically they would result in "competing" strain terms). Is there a way I can directly evolve the strain in both Tensor Mechanics and Phase Field instead of any underlying material properties?

**Reply by RobMacka** [reply]
Essentially the free energy density includes both magnetic variables and strain tensor components simulataneously, which can't be separated in any sensible way.

**Reply by GiudGiud** [reply]
So for the free energy you can write it as a parsed function so it's fairly easy to make complicated dependencies.

For having two different sources of stress you can use this material maybe?
https://mooseframework.inl.gov/source/materials/ComputeMultipleInelasticStress.html
This is not my area of expertise, but I think it s been done to combine strain sources
